docs-web/src/main/java/com/sismics/docs/rest/resource/FileResource.java

```diff
@@ -738,26 +773,60 @@ public Response zip(
      * @param shareId Share ID
      * @return File
      */
-    private File findFile(String fileId, String shareId) {
+    private File findFile(String fileId, String shareId, boolean jsonResponse) {
         FileDao fileDao = new FileDao();
         File file = fileDao.getFile(fileId);
         if (file == null) {
             throw new NotFoundException();
         }
+        checkFileAccessible(shareId, file, jsonResponse);
+        return file;
+    }
+
+
+    /**
+     * Find a list of giles with access rights checking.
+     *
+     * @param filesIds Files IDs
+     * @return List<File>
+     */
+    private List<File> findFiles(List<String> filesIds) {
+        FileDao fileDao = new FileDao();
+        List<File> files = fileDao.getFiles(filesIds);
+        for (File file : files) {
+            checkFileAccessible(null, file, false);
+        }
+        return files;
+    }
 
+    /**
+     * Check if a file is accessible to the current user
+     * @param shareId Share ID
+     * @param file
+     */
+    private void checkFileAccessible(String shareId, File file, boolean jsonResponse) {
         if (file.getDocumentId() == null) {
             // It's an orphan file
             if (!file.getUserId().equals(principal.getId())) {
                 // But not ours
-                throw new ForbiddenClientException();
+                if (jsonResponse) {
```

Comment:
I'm not fan of this, is there a way to always throw the same exception?

Reply:
I wasn't too happy about it either, after enough digging in the doc I found you could use a `MessageBodyWriter` to fix this case, which means the solution is even generic. Whet do you think about it?